### ISSUE TITLE
Predefined type FormattableStringFactory error not shown by Visual Studio

### ISSUE THREAD

**Issue by AmadeusW** [dotnet/roslyn#7093]
I'm trying to interpolate a string using invariant culture, and I'm getting `Predefined type 'System.Runtime.CompilerServices.FormattableStringFactory' is not defined or imported`. The error is only visible in the Output > Build box. Visual Studio doesn't show the error in the user interface.

``` csharp
var pathDefinition = formatInvariant($"M {point1.X} {point1.Y} A {OUTER_RADIUS} {OUTER_RADIUS} 0 0 1 {point2.X} {point2.Y} L {point3.X} {point3.Y} A {INNER_RADIUS} {INNER_RADIUS} 0 0 0 {point4.X} {point4.Y} z");
// ...

private static string formatInvariant(IFormattable format)
{
    return format.ToString(null, System.Globalization.CultureInfo.InvariantCulture);
}
```

Line 30 in the error refers to `var pathDefinition . . .`

```
1>C:\Users\amade\Documents\GitHub\TouchVS\src\CodeConnect.Touch\Helpers\TouchControlShapeFactory.cs(30,50,30,222): error CS0518: Predefined type 'System.Runtime.CompilerServices.FormattableStringFactory' is not defined or imported
```

Screenshot:
![img](https://i.gyazo.com/9c9821b0cda1d0ecf444f0b542371253.png)

**Comment by AmadeusW** [comment]
[I see that I need to use .NET framework 4.6](https://stackoverflow.com/questions/32076823/string-interpolation-in-visual-studio-2015-and-iformatprovider-ca1305). This solved my issue.

Is there a way to get Visual Studio to display this error?

**Comment by gafter** [comment]
We happen to report this error during code generation, which is too late to be useful to you. Perhaps we should move it earlier.

**Comment by gafter** [comment]
@dotnet/roslyn-analysis When an errors occurs in building, and the Error List is set to "Build + Intellisense", this error is not being shown in the Error List. That sounds like a bug in the Error List. Reassigning to the Analyzers team.

**Comment by heejaechang** [comment]
this seems another error live GetDiagnostics won't report. we need to add this error to whitelist.
